code/messaging/src/phone/java/com/adobe/marketing/mobile/MessageDelegate.java

```diff
@@ -0,0 +1,240 @@
+  Copyright 2021 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile;
+
+import static com.adobe.marketing.mobile.MessagingConstants.LOG_TAG;
+
+import android.webkit.ValueCallback;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+
+import com.adobe.marketing.mobile.services.ServiceProvider;
+import com.adobe.marketing.mobile.services.ui.AEPMessage;
+import com.adobe.marketing.mobile.services.ui.FullscreenMessage;
+import com.adobe.marketing.mobile.services.ui.FullscreenMessageDelegate;
+import com.adobe.marketing.mobile.services.ui.MessageSettings;
+import com.adobe.marketing.mobile.services.ui.UIService;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class is the Messaging extension's internal implementation of the {@link FullscreenMessageDelegate}.
+ * Additionally, this class handles the dispatching of Experience Edge tracking events as well as opening URL's
+ * and loading any javascript code for the {@link Message} class.
+ */
+public class MessageDelegate implements FullscreenMessageDelegate {
+    private final static String SELF_TAG = "MessageDelegate";
+    private final static String AMPERSAND = "&";
+    private final static String EXPECTED_JAVASCRIPT_PARAM = "js=";
+    private final static String ADOBE_DEEPLINK = "adb_deeplink";
+    // public properties
+    public String messageId;
+    public boolean autoTrack = true;
+    // internal properties
+    MessagingInternal messagingInternal;
+    Map<String, Object> details;
+
+    /**
+     * Dispatch tracking information via a Messaging request content event.
+     *
+     * @param interactionType {@code String} containing the interaction which occurred
+     */
+    public void track(final String interactionType) {
+        if (StringUtils.isNullOrEmpty(interactionType)) {
+            Log.debug(LOG_TAG,
+                    "%s - Unable to record a message interaction - interaction string was null or empty.", SELF_TAG);
+            return;
+        }
+
+        final HashMap<String, Object> eventData = new HashMap<>();
+        eventData.put(MessagingConstants.EventDataKeys.Messaging.TRACK_INFO_KEY_EVENT_TYPE, MessagingConstants.EventDataKeys.Messaging.IAMDetailsDataKeys.EventType.INTERACT);
```

Comment:
The constant names are really big. Probably we can have another class inside `Messaging` to shorten them.
i.e `MessagingConstants.EventDataKeys.Messaging.TrackInfo.EVENT_TYPE`. What is you opinion on this? Will it improve readability.

Reply:
yeah i can shorten them for readability